### ISSUE TITLE
Generating blank pdf

### ISSUE THREAD

**Issue by swaps19** [mileszs/wicked_pdf#567]
The pdf generation behaves randomly.
I have faced this many times while using `wicked_pdf` that sometimes the pdf that is generated is blank other than the cover page and if I try again to generate the same pdf it generates as expected. I am wondering why is this kind of random behavior in operations? The main issue where this thing is causing problem is that I am also generating bulk pdfs; in that case it is not possible to reload or regenerate the pdf as its very time consuming operation to regenerate all pdfs that were generated in bulk.

Is there any explanation why this random behavior in pdf generation?

**Comment by txssseal** [comment]
this issue should be closed @swaps19

**Comment by mindaslab** [comment]
I am getting blank pdf (just 1 page) when I try to add header to it. Wondering if its the same issue.

**Comment by swaps19** [comment]
@mindaslab for `header` & `footer` stuff, please check the `HTML` page you have written. It should have proper syntax(including opening & closing tags) with `<!DOCTYPE>`

**Comment by walidvb** [comment]
Also, check that you have margins set on your page. Without margin I'd get a completely white page(with no content at all, tried to search text). Margins made everything work!

**Comment by unixmonkey** [comment]
@swaps19 Did you figure out your issue here? What was the resolution?

**Comment by swaps19** [comment]
@unixmonkey I'm not sure. Its been a long time since I had raised this issue. As far as I can remember I had done some workaround but unfortunately I'm not able to post it as it is with my previous organization. I presume there might have been many updates included in the gem, so this issue may not be valid now. If anyone faces the same, they can definitely reopen this one.
I just closed this as the issue list is already high and it might be a trouble for maintainers.
